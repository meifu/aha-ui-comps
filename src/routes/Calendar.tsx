import type {} from '@mui/x-date-pickers/themeAugmentation';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { StaticDatePicker } from '@mui/x-date-pickers';

import { createTheme, ThemeProvider } from '@mui/material/styles';


const theme = createTheme({
  components: {
    MuiStaticDatePicker: {
      // styleOverrides: {},
    },
    MuiDatePicker: {
      defaultProps: {},
      // styleOverrides: {},
    },
    MuiSelect: {
      styleOverrides: {

      }
    },
  }
})

export default function Calendar() {
  const dayOfWeekFormatter = (day: string, date: string) => {
    return day;
  };

  return (
    <div className="max-w-md">
      <LocalizationProvider dateAdapter={AdapterDayjs}>
        <ThemeProvider theme={theme}>
          <StaticDatePicker
            sx={{
              width: '320px',
              backgroundColor: '#1B1B1B',
              boxShadow: '4px 4px 20px 0 rgba(0, 0, 0, .3)',
              '& .css-1u23akw-MuiButtonBase-root-MuiPickersDay-root': {
                color: '#fff',
                fontSize: '14px',
              },
              '& .css-1u23akw-MuiButtonBase-root-MuiPickersDay-root:hover': {
                color: '#1B1B1B',
                backgroundColor: '#fff',
              },
              '& .css-rhmlg1-MuiTypography-root-MuiDayCalendar-weekDayLabel': {
                color: '#929292',
                fontSize: '11px',
              },
              '& .css-jgls56-MuiButtonBase-root-MuiPickersDay-root:not(.Mui-selected)': { // old selected
                backgroundColor: '#1B1B1B',
                color: '#fff',
                borderColor: '#00A3FF',
              },
              '& .css-1vooibu-MuiSvgIcon-root': {
                color: '#fff',
              },
              '& .css-1e6y48t-MuiButtonBase-root-MuiButton-root': {
                color: '#fff',
                fontSize: '14px',
                fontWeight: '600',
              },
              '& .css-1u23akw-MuiButtonBase-root-MuiPickersDay-root.Mui-selected': { // selected
                backgroundColor: '#00A3FF',
                border: 'solid',
                borderColor: '#00A3FF',
              },
              '& .css-1u23akw-MuiButtonBase-root-MuiPickersDay-root:focus.Mui-selected': { // focus
                backgroundColor: '#00A3FF',
              },
              '& .css-sv1lee-MuiPickersLayout-root .css-1u23akw-MuiButtonBase-root-MuiPickersDay-root:hover': {
                color: '#fff',
              },
              '& .css-18wn1yb-MuiButtonBase-root-MuiPickersDay-root': { // days outside current month
                color: '#929292',
              },
              '& .MuiPickersArrowSwitcher-root.css-9reuh9-MuiPickersArrowSwitcher-root': { // month switcher arrows
                width: '100%',
                justifyContent: 'space-between',
              },
              '& .css-cyfsxc-MuiPickersCalendarHeader-labelContainer': { // current month year
                position: 'absolute',
                left: 'calc(50% - 40px)',
              },
              '& .MuiPickersCalendarHeader-root': {
                position: 'relative',
              }
            }}
            dayOfWeekFormatter={dayOfWeekFormatter}
            showDaysOutsideCurrentMonth={true}
          />
        </ThemeProvider>
      </LocalizationProvider>
    </div>
  );
};