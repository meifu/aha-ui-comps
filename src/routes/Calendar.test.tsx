import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Calendar from './Calendar';

describe('Component: Calendar', () => {
  it('render Calendar', () => {
    render(<Calendar />);
    // const cancelBtn = screen.getByText('Cancel');
    // expect(cancelBtn).toBeInTheDocument();
    // const okBtn = screen.getByText('OK');
    // expect(okBtn).toBeInTheDocument();
  });

  it('initial snapshot', () => {
    const tree = renderer.create(<Calendar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
