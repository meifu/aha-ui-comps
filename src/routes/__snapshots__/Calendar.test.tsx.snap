// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Calendar initial snapshot 1`] = `
<div
  className="max-w-md"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-clzcj4-MuiFormLabel-root-MuiInputLabel-root"
    htmlFor="birthday"
    size="small"
  >
    Birthday
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-7bdm4b-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input MuiOutlinedInput-input css-p51h6s-MuiInputBase-input-MuiOutlinedInput-input"
      name="birthday"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="mm/dd/yyyy"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
